packages/app/src/systems/Settings/services/abi.ts

```diff
@@ -0,0 +1,47 @@
+import type { AbiMap } from '@fuel-wallet/types';
+
+import { db } from '~/systems/Core/utils/database';
+
+export type AbiInputs = {
+  addAbi: {
+    data: AbiMap;
+  };
+};
+
+export class AbiService {
+  static async addAbi(input: AbiInputs['addAbi']) {
+    return db.transaction('rw', db.abis, async () => {
+      const abiMap = await AbiService.getAbiMap();
+      const dataToAdd = Object.keys(input.data)
+        .map((key) => ({
+          contractId: key,
+          abi: input.data[key],
+        }))
+        .filter(({ contractId }) => !abiMap[contractId]);
+      await db.abis.bulkAdd(dataToAdd);
```

Comment:
do we need bulkAdd is we are only adding 1 abi at a time per call.

Reply:
user can add multiple as abiMap can be passed with multiple abis

```
addAbi({
   [contractId1]: abi1,
   [contractId2]: abi2,
});
```